docs/content/connector/connector_protocol.md

```diff
@@ -0,0 +1,657 @@
+   This gives the external connector the possibility to signalize faulty configuration parameters (e.g. entered by developer).
+
+* 5: ERROR
+
+   Set by connctd as soon as the external connector service replied with an unexpected status code or when the response is malformed.
+   In this case connctd responds with its own state details which should give external connector developers information on how to fix the problems.
+
+* 6: INTERNAL_ERROR
+
+   Set by connctd due to an internal issue.
+   Leads to the cancellation of the installation.
+
+* 7: REMOVAL_ONGOING
+
+   Set by connctd as soon as an installation was marked for removal.
+
+#### Installation step types
+
+* 1: TEXT
+
+   The corresponding value is interpreted as text.
+
+* 2: MARKDOWN
+
+   The corresponding value is interpreted as markdown.
+
+* 3: REDIRECT
+
+   The corresponding value is expected to be an URL.
+   A link to the URL will be shown in the [Developer Center](https://devcenter.connctd.io/).
+
+### Instance callback
+
+While an App can only install a connector once, it can create multiple instantiations of the installed connector.
+As soon as a connector is installed, instances of it can be spawned for different end users (separated by external subject ids).
+Whenever a connector installation is instantiated on the connctd platform a POST request is send against the *instanceCallbackURL* specified in the connector publication.
+When a instantiation is removed a DELETE request is send against *instanceCallbackURL/-installationId-*.
+
+### New instantiation
+
+```http
+POST /-instanceCallbackUrl HTTP/1.1
+Content-Type: application/json
+Signature: -base64encodedSignature-
+
+{
+    "id": "-instanceId-",
+    "installation_id": "-installationId",
+    "token": "-instanceToken-",
+    "state": 1,
+    "configuration": [
+        {
+            "id": "-instantiationParameter-",
+            "value": "-parameterValue-"
+        }
+    ]
+}
+```
+
+See [installation documentation](#new-installation) for information on the signature.
+
+#### Payload
+
+`id` is the ID of the new instance generated by the conctd platform.
+It is used for further communication with the platform and should be stored by the connector.
+
+`token` is the ***instance token*** that is generated by the connctd platform for this specific instance.
+It is used to authenticate requests towards the connctd platform and should be saved by the connector.
+
+`state` is the current state of the instance.
+It is set to **1** (**INITIALIZED**) and can have one of the values described [below](#instance-states).
+
+`configuration` will contain all configuration parameters provided by the instantiation.
+It will contain at least the required parameters and all parameters are type checked and validated according to the requirements specified in the connector publication before sending the request.
```

Comment:
```suggestion
It will contain at least the required parameters specified in the originating connector publication. Additionally all parameters are type checked and validated beforehand.
```